frontend/src/app/component/auth-component/login.component.ts

```diff
@@ -0,0 +1,107 @@
+import { Component, inject } from '@angular/core';
+import { FormBuilder, ReactiveFormsModule, Validators } from '@angular/forms';
+import { RouterLink, Router } from '@angular/router';
+import { NgIf } from '@angular/common';
+import { AuthService } from '../../services/auth-services/auth.service';
+import { ToastService } from '../../services/auth-services/toast.service';
+import { LoadingService } from '../../services/auth-services/loading.service';
+import { InputTextModule } from 'primeng/inputtext';
+import { PasswordModule } from 'primeng/password';
+import { ButtonModule } from 'primeng/button';
+import { HttpErrorResponse } from '@angular/common/http';
+
+@Component({
+  standalone: true,
+  selector: 'app-login',
+  imports: [ReactiveFormsModule, RouterLink, NgIf, InputTextModule, PasswordModule, ButtonModule],
+  template: `
+    <div class="auth-container">
+      <section class="auth-card">
+        <h1>Login</h1>
+        <p>Access your account.</p>
+
+        <form [formGroup]="form" (ngSubmit)="submit()">
+          <div class="field">
+            <label>Email</label>
+            <input type="email" pInputText formControlName="email" />
+            <small *ngIf="email.touched && email.invalid">Please enter a valid email.</small>
+          </div>
+
+          <div class="field">
+            <label>Password</label>
+            <p-password
+              formControlName="password"
+              [feedback]="false"
+              [toggleMask]="true"
+            ></p-password>
+            <small *ngIf="password.touched && password.hasError('required')"
+              >Password is required.</small
+            >
+          </div>
+
+          <button
+            pButton
+            type="submit"
+            [disabled]="loading.active() || form.invalid"
+            [label]="loading.active() ? 'Signing in…' : 'Sign in'"
+          ></button>
+        </form>
+
+        <p class="switch-link">
+          Don’t have an account?
+          <a routerLink="/signup">Create one</a>
+        </p>
+      </section>
+    </div>
+  `,
+  styleUrls: ['./auth-styles.css'],
+})
+export class LoginComponent {
+  private fb = inject(FormBuilder);
```

Comment:
```suggestion
  private formBuilder = inject(FormBuilder);
```
use expressive names, so that you can easily identify what's missing lately